formatted_text.md

```diff
@@ -0,0 +1,129 @@
+# Formatted text
+
+In various fields of the index, authors may write formatted text.
+
+For best interoperability and simplicity, we use an XML-based syntax which is similar to a subset of HTML.
+
+Motivating example:
+```
+This is an <b>example</b> mod. The word "<b>example</b>" is written in bold.
+
+<h1>Blocks in my mod</h1>
+<ul>
+<li><b>Dostuffulator</b> - it does stuff.</li>
+<li><b>Nitroglycerine</b> - it <big>explodes</big> when you click on it!</li>
+</ul>
+
+<h1>Recipes</h1>
+<b>Dostuffulator</b>:
+<img id="1234"/>
+<b>Nitroglycerine</b>:
+<img id="2345"/>
+```
+
+Since XML likes to work on documents, not fragments, the whole text is implicitly surrounded by `<?xml version="1.0" encoding="utf-8"?><root>...</root>`.
+The point is that this boilerplate should not be included in every formatted text field.
+
+# Inline and block contexts
+
+Formatted text occurs in two contexts: inline contexts and block contexts.
+
+Inline contexts include summary blurbs, names of people, and generally short pieces of content that are embedded in a larger piece.
+Inline contexts only allow simple per-character formatting, such as bold or coloured text.
+
+Block contexts are generally complete documents with their own structure, such as project descriptions.
+Block contexts allow document structure elements, such as headings, lists, and large elements such as images.
+
+*Unless specified otherwise, all elements are only valid in block contexts*
+
+# Elements
+
+## Simple formatting
+
+Allowed in inline contexts.
+
+* `<b>...</b>` - text within is bold
+* `<i>...</i>` - text within is italic
+* `<u>...</u>` - text within is underlined
+* `<s>...</s>` - text within is struck through
+* `<tt>...</tt>` - text within is monospace. Some viewers may also change the background and add a border, to indicate a code block.
+* `<col c="#rrggbb">...</col>` - coloured text. Some viewers may alter the colour, if it conflicts with the background for example.
+* `<a href="...">...</a>` - hyperlink. URL must be absolute. Links might not be allowed in all contexts.
+
+## Block formatting
+
+* `<h1>...</h1>` up to h6 - heading levels
+* `<ul>...</ul>` - bullet list, list items are in `<li>...</li>`
+* `<ol>...</ol>` - numbered list, list items are in `<li>...</li>`
+* `<fold title="Recipes">...</fold>` - a foldable content section (a "spoiler"). `title` is optional - if omitted, it may default to something like "click here to open" (viewer-dependent).
+* Whitespace - double-newline should be similar to HTML's `<p>` and single newline should be similar to HTML's `<br>`
+
+(TODO: should we have explicit paragraph and newline tags?)
+
+Not that even paragraph breaks and line breaks are not allowed in inline context.
+
+## Embedded content
+
+* `<media id="..."/>` - display an image or video from the project. All content within the element is ignored (even `<vis>`). (TODO: what does the ID refer to?)
```

Comment:
Is this supposed to be a replacement for `<img>`? Is it tied to specific images/videos for the project (e.g. in the `media` list)?

Reply:
That's the idea. Those items will need to have an ID. I wasn't sure whether URLs should be allowed, because they *will* get broken and then the description isn't self-contained. (putting them in the media list allows them to be downloaded at any time)

Reply:
I see. I think for URLs it might be a better idea to use traditional elements (`<img>`, `<video>`). It makes it easier to convert HTML to Formatted Text, and it creates a clear distinction between project media and other media.

Reply:
Note that youtube is also not self-contained. Should it be allowed?

Reply:
I don’t think it should be allowed for `<media>‘. I think YouTube should use a tag specifically for external content (`<embed>` or `<youtube>` depending on which one is finalized in the spec).